moto/rds/models.py

```diff
@@ -619,12 +622,11 @@ def __init__(self, **kwargs: Any):
             self.availability_zone = f"{self.region_name}a"
         self.multi_az = kwargs.get("multi_az")
         self.db_subnet_group_name = kwargs.get("db_subnet_group_name")
+        self.db_subnet_group = None
```

Comment:
setting this first instead of in the `else` branch means that mypy figures out it's an Optional type automatically